common/src/Model/ParallelTexCoordSystem.cpp

```diff
@@ -102,18 +102,14 @@ namespace TrenchBroom {
             m_yAxis = rot * m_yAxis;
         }
 
-        void ParallelTexCoordSystem::doTransform(const Plane3& oldBoundary, const Mat4x4& transformation, BrushFaceAttributes& attribs, bool lockTexture, const Vec3& oldInvariant) {
+        void ParallelTexCoordSystem::doTransform(const Plane3& oldBoundary, const Mat4x4& origTransformation, BrushFaceAttributes& attribs, bool lockTexture, const Vec3& oldInvariant) {
 
-            // compute the new texture axes
-            const Vec3 offset = transformation * Vec3::Null;
-            m_xAxis           = transformation * m_xAxis - offset;
-            m_yAxis           = transformation * m_yAxis - offset;
-            assert(!m_xAxis.nan());
-            assert(!m_yAxis.nan());
-
-            if (!lockTexture || attribs.xScale() == 0.0f || attribs.yScale() == 0.0f)
+            if (attribs.xScale() == 0.0f || attribs.yScale() == 0.0f)
                 return;
             
+            // when texture lock is off, don't compensate for the translation part of the transformation
+            const Mat4x4 transformation = lockTexture ? origTransformation : stripTranslation(origTransformation);
```

Comment:
I'd prefer to use a name like `actualTransformation` or `effectiveTransformation`here.